graph/src/components/metrics/stopwatch.rs

```diff
@@ -0,0 +1,169 @@
+use crate::prelude::*;
+use std::collections::HashMap;
+use std::sync::{atomic::AtomicBool, atomic::Ordering, Mutex};
+use std::time::Instant;
+
+/// This is a "section guard", that closes the section on drop.
+pub struct Section {
+    id: String,
+    stopwatch: StopwatchMetrics,
+}
+
+impl Section {
+    /// A more readable `drop`.
+    pub fn end(self) {}
+}
+
+impl Drop for Section {
+    fn drop(&mut self) {
+        self.stopwatch
+            .end_section(std::mem::replace(&mut self.id, String::new()))
```

Comment:
What's the `std::mem::replace` trick here doing?

Reply:
Simulating a move to prevent cloning `self.id`. We can't actually move because drop doesn't take `self`.